### PR TITLE
CVE-2010-1157

### PR DESCRIPTION
**Pull request by Scar56**
filling out cve-2010-1157

### CODE REVIEW

**Comment by spc7641** [comment]
For the description section give additional details, currently it is confusing to read. Make sure that you write what happened in between the vcc and the fix in the interesting commits section if you found no interesting commits. For the mistakes section give information on why this vulnerability occurred, as well as how their mitigation actually worked or did not. Make sure for all paragraph sections you use proper capitalization and give context to the writing.  

Upvotes: 1

File: cves/CVE-2010-1157.yml [hunk 1]
```diff
@@ -43,31 +43,30 @@ description_instructions: |
...
   Please put the Git commit SHA in "commit" below, and any notes about how this
   was discovered in the "note" field.
   Refer to our instructions on how to find a Git SHA from an SVN revision.
 fixes:
-   - commit: 78e45a225c5c5564945191f7e421f8859885d63b
-     note: SVN rev 936541, from the Tomcat website.
-   - commit:
-     note:
+   - commit: a4de93b2aaf74fff7c1e194f7ce3d2228a9680b8
+     note: SVN rev 936539, from the Tomcat website.
```
Comment (ADAdebomi) [inline cves/CVE-2010-1157.yml]: Possibly add a little description of what happens in the fix? what changes were made

File: cves/CVE-2010-1157.yml [hunk 2]
```diff
@@ -43,31 +43,30 @@ description_instructions: |
...
 vccs:
-  - commit:
-    note:
-  - commit:
-    note:
+  - commit: a84fabcbc6fee8a69253ad92a304b4718e96a7c9
+    note: SVN rev 389146, file added
```
Comment (ADAdebomi) [inline cves/CVE-2010-1157.yml]: Maybe explain what changed in this commit that led to the vulnerability's creation

File: cves/CVE-2010-1157.yml [hunk 3]
```diff
@@ -43,31 +43,30 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description:
+description: |
+    might allow remote attackers to discover the server's hostname or IP address
+    by sending a request for a resource that requires one of two specific
+    authentication levels
```
Comment (ADAdebomi) [inline cves/CVE-2010-1157.yml]: Description could be a lot clearer, doesn't state how the vulnerability works

File: cves/CVE-2010-1157.yml [hunk 4]
```diff
@@ -202,4 +205,6 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer:
+  answer: |
+      certain response headers contain confidential data (the server IP
```
Comment (ADAdebomi) [inline cves/CVE-2010-1157.yml]: Possibly mitigation as to how you think this mistake could have been avoided

File: cves/CVE-2010-1157.yml [hunk 5]
```diff
@@ -10,27 +10,27 @@ CWE_instructions: |
...
   set to true.
-curated: false
+curated: true
 
 reported_instructions: |
   Was there a date that this vulnerability was reported to the team? You can
   find this in changelogs, blogs, bug reports, or perhaps the CVE data.
   Please enter your date in YYYY-MM-DD format.
-reported:
+reported: 2010-03-29
```
Comment (ADAdebomi) [inline cves/CVE-2010-1157.yml]: Upvote: 1

**Pull request merged**